packages/main/src/themes/ValueStateMessage.css

```diff
@@ -87,6 +90,71 @@
 }
 
 .ui5-valuestatemessage-popover::part(header),
-.ui5-suggestions-popover-with-value-state-header::part(header) {
+.ui5-popover-with-value-state-header::part(header),
+.ui5-popover-with-value-state-header-phone::part(header) {
 	margin-bottom: 0;
 }
+
+.ui5-popover-with-value-state-header .ui5-valuestatemessage-root {
+	padding: var(--_ui5_value_state_message_padding);
+}
+
+.ui5-popover-with-value-state-header-phone .ui5-valuestatemessage-root {
+	padding: var(--_ui5_value_state_message_padding_phone);
+	width: 100%;
+}
+
+.ui5-popover-with-value-state-header-phone .ui5-input-value-state-message-icon {
+	left: var(--_ui5_value_state_message_icon_offset_phone);
+}
+
+.ui5-popover-with-value-state-header-phone .ui5-valuestatemessage-header {
+	position: relative;
+	flex: none;
+	top: 0;
+	left: 0;
+}
+
+.ui5-popover-with-value-state-header-phone::part(content) {
+	padding: unset;
+	overflow: hidden;
+	display: flex;
+	flex-direction: column;
+}
+
+.ui5-popover-with-value-state-header-phone .ui5-valuestatemessage-root + [ui5-list] {
+	overflow: auto;
+}
+
+/* ValueState Header bottom border color in Responsive Popover and Dialog */
+[ui5-responsive-popover] .ui5-valuestatemessage--error {
+	box-shadow: var(--_ui5_value_state_header_box_shadow_error);
+}
+
+[ui5-responsive-popover] .ui5-valuestatemessage--information {
+	box-shadow: var(--_ui5_value_state_header_box_shadow_information);
+}
+
+[ui5-responsive-popover] .ui5-valuestatemessage--success {
+	box-shadow: var(--_ui5_value_state_header_box_shadow_success);
+}
+
+[ui5-responsive-popover] .ui5-valuestatemessage--warning {
+	box-shadow: var(--_ui5_value_state_header_box_shadow_warning);
+}
+
+[ui5-responsive-popover].ui5-popover-with-value-state-header .ui5-valuestatemessage--information:has(+[ui5-list]:empty) {
+	box-shadow: unset;
+}
+
+[ui5-responsive-popover].ui5-popover-with-value-state-header .ui5-valuestatemessage--warning:has(+[ui5-list]:empty) {
+	box-shadow: unset;
+}
+
+[ui5-responsive-popover].ui5-popover-with-value-state-header .ui5-valuestatemessage--error:has(+[ui5-list]:empty) {
+	box-shadow: unset;
+}
+
+[ui5-responsive-popover].ui5-popover-with-value-state-header .ui5-valuestatemessage--success:has(+[ui5-list]:empty) {
+	box-shadow: unset;
+}
```

Comment:
I will add only 1 rule to unset the box-shadow in the very specific case it is not needed and use the .ui5-valuestatemessage-root class. Otherwise the selectors when to set the styles will become even more complex than the suggested alternative